architecture.md

```diff
@@ -0,0 +1,685 @@
+# Kata Containers Architecture
+
+* [Overview](#overview)
+    * [Hypervisor](#hypervisor)
+      * [Assets](#assets)
+        * [Guest kernel](#guest-kernel)
+        * [Root filesystem image](#root-filesystem-image)
+    * [Agent](#agent)
+    * [Runtime](#runtime)
+        * [Configuration](#configuration)
+        * [Significant commands](#significant-commands)
+            * [create](#create)
+            * [start](#start)
+            * [exec](#exec)
+            * [kill](#kill)
+            * [delete](#delete)
+    * [Proxy](#proxy)
+    * [Shim](#shim)
+    * [Networking](#networking)
+    * [Storage](#storage)
+    * [Kubernetes Support](#kubernetes-support)
+        * [Problem Statement](#problem-statemem)
+        * [OCI Annotations](#oci-annotations)
+        * [Generalization](#generalization)
+        * [Mixing VM based and namespace based runtimes](#mixing-vm-based-and-namespace-based-runtimes)
+* [Appendices](#appendices)
+    * [DAX](#dax)
+    * [Previous Releases](#previous-releases)
+    * [Resources](#resources)
+
+## Overview
+
+This is an architectural overview of Kata Containers, based on the 1.0 release.
+
+The two primary deliverables of the Kata Containers project are a container runtime
+and a CRI friendly library API.
+
+The [Kata Containers runtime (kata-runtime)](https://github.com/kata-containers/runtime)
+is compatible with the [OCI](https://github.com/opencontainers) [runtime specification](https://github.com/opencontainers/runtime-spec)
+and therefore works seamlessly with the
+[Docker\* Engine](https://www.docker.com/products/docker-engine) pluggable runtime
+architecture. It also supports the [Kubernetes\* Container Runtime Interface (CRI)](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/apis/cri/v1alpha1/runtime)
+through the [CRI-O\*](https://github.com/kubernetes-incubator/cri-o) and
+[CRI-containerd\*](https://github.com/containerd/cri) implementation. In other words, you can transparently
+select between the [default Docker and CRI shim runtime (runc)](https://github.com/opencontainers/runc)
+and `kata-runtime`.
+
+![Docker and Kata Containers](arch-images/docker-kata.png)
+
+`kata-runtime` creates a QEMU\*/KVM virtual machine for each container or pod
+the Docker engine or Kubernetes' `kubelet` creates.
+
+The container process is then spawned by an
```

Comment:
I think the following would improve this section slightly:

> The container process is then spawned by [`kata-agent`](https://github.com/kata-containers/agent), an agent process running as a daemon inside the virtual machine. `kata-agent` runs a gRPC server in the guest using a virtio serial interface which QEMU exposes as a serial device on the host. `kata-runtime` uses a gRPC protocol to communicate with the agent. This protocol allows the runtime to send container management commands to the agent. The protocol is also used to pass I/O streams (`stdout`, `stderr`, `stdin`) between the guest and the Docker Engine.

Full diff:
- https://github.com/kata-containers/documentation/wiki/temporary:-kata-1.0-arch-overview/_compare/b824e764fef9d0843df09b5f9ee8b2410e357989...4fea8292cb016ec2dee462fe8e359fb8f6ee436e

Reply:
done